python/ray/experimental/sgd/util.py

```diff
@@ -136,3 +162,20 @@ def chrome_trace_format(self, filename):
     b.end("b1")
     a.merge(b)
     a.chrome_trace_format("test.json")
+
+    import threading
+    import random
+    print("Testing file lock, this will take a few seconds...")
+    lock_id = random.randint(0, 100000)
+    lock_path = "/tmp/test" + str(lock_id) + ".lock"
+    lock1 = SoftFileLock(lock_path)
+
+    def try_hold_lock():
+        lock2 = SoftFileLock(lock_path)
+        lock2.acquire()
+        time.sleep(10.0)
+        lock2.release()
+
+    threading.Thread(target=try_hold_lock).start()
```

Comment:
For the test, maybe run 100 of them in separate threads (where each one sleeps for 0.1s) and make sure the whole thing takes at least 10s or something like that?